Removing the legacy builder

escopecz opened the pull request:
<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 5.x)
* a.b for any bug fixes (e.g. 4.4, 5.1)
* c.x for any features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 5.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 <!-- Use emojis to indicate positive (green) or negative (red) for each item in the table. -->
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🟢
| Automated tests included?              | 🔴 (cannot write tests for deleted code)
| Related user documentation PR URL      | mautic/user-documentation#... <!-- required for new features -->
| Related developer documentation PR URL | mautic/developer-documentation-new#... <!-- required for developer-facing changes -->
| Issue(s) addressed                     | Fixes #... <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

## Description

It's time to remove the Legacy builder that funny enough I mostly built. It got replaced by the GrapesJS builder many years ago and only few are still preferring the legacy builder nowadays. It got broken during the latest UI upgrades and not many notice (I saw 1 discussion on the topic) and there was no one who would care to fix it. So it's time to say goodbye and make Mautic a little leaner, simpler and faster.

This is what happens when you disable the GrapesJS plugin and try to use the email or page builder:
<img width="836" alt="Screenshot 2025-01-26 at 19 52 03" src="https://github.com/user-attachments/assets/8d3a34cc-f9ca-413b-b49f-830a78326832" />

Todo:
- [x] Regenerate css
- [x] Go through removed twig templates and search for translation strings if we still need those
- [x] Fix CI

<!--
Please write a short README for your feature/bugfix. This will help people understand your PR and what it aims to do. If you are fixing a bug and if there is no linked issue already, please provide steps to reproduce the issue here.
-->
<!-- Remove HTML comment markup below to use the table for screenshots when relevant. -->
<!--
| Before                                 | After
| -------------------------------------- | ---
|                                        | 
-->


---
### 📋 Steps to test this PR:

<!--
This part is crucial. Take the time to write very clear, annotated and step by step test instructions, because testers may not be developers.
-->
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Test that the GrapesJS builder works as before in Landin Pages and Emails
3. Test that the preference center still works. Both when using a preference center landing page and the default preference center.
4. Test that nothing really breaks when you disable the GrapesJS plugin but you won't be able to launch the builder and a helpful message will show up instead.

<!--
If you have any deprecations and backwards compatibility breaks, list them here along with the new alternative.
-->

escopecz committed:
Removing the legacy builer - slots and sections

escopecz committed:
Property $security is never read, only written

escopecz committed:
The slot function no longer exist in Twig

escopecz committed:
Simplified test

escopecz committed:
Fixing preference center checkbox

escopecz committed:
Returning methods needed by the GrapesJS Builder

escopecz committed:
Adding back used translation

escopecz committed:
Using tokens in the test instead of slots

escopecz committed:
Optiizing the prefcenter JS

escopecz committed:
Fixing the aria-hidden JS error message

escopecz committed:
Fixing the right param count

escopecz committed:
These 2 tests are causing an infinite loop it seems?

escopecz committed:
CS fixes

escopecz committed:
Rebuilding CSS from LESS

escopecz committed:
Returning back the builder-active styling needed for GrapesJS to show up

escopecz committed:
Enabling the problematic tests back

matbcvo commented on app/bundles/CoreBundle/Assets/js/4.builder.js:
```diff
@@ -1,184 +1,5 @@
-
-    return elementBottom > viewportTop && elementTop < viewportBottom;
+Mautic.launchBuilder = function () {
+    alert('Please enable the GrapesJS builder plugin to use this feature. Or other builder plugin.');
```
```suggestion
    alert('Please enable the GrapesJS builder plugin (or another builder plugin) to use this feature.');
```
This needs to be rewritten. I'm suggesting the above change - what do you think? @RCheesley

RCheesley replied:
I think that's great, thanks @matbcvo

escopecz committed:
Fixing the focus items builder.

Some CSS styles that seem to be related to the legacy builder must be returned back

escopecz committed:
wording update

Co-authored-by: Martin Vooremäe <martin.vooremae@gmail.com>

matbcvo reviewed (approved):
The code changes look good to me. I’ve tested the PR and found no issues.

RCheesley committed:
Merge branch '6.x' into removing-legacy-builder

RCheesley commented:
When I tried to test this by opening an old email (the test one from sample data) which used the blank theme, I am greeted with a partly blank screen with no builder or editor.

I think for these old theme emails we should perhaps be opening them in the GrapesJS builder as an HTML email?

Here's a screencast: https://app.screencastify.com/v3/watch/Nmlm5epI7LeAY7zBEDdn

The blank theme _should_ support the GrapesJS builder however it seems like it's loading in the old one:

![Themes-Mautic-02-05-2025_11_30_AM](https://github.com/user-attachments/assets/70519045-815b-4d2f-9b8a-414a54d7b191)

RCheesley reviewed (changes requested):
Sorry forgot to leave as a RQ review!

escopecz commented:
@RCheesley I cannot reproduce:

https://github.com/user-attachments/assets/fe5ae3f4-7ced-4db7-871c-07643ff776bd

Please make sure you've run `npm ci` and `bin/console mautic:assets:generate` and also clear assets in your browser. Perhaps there is something cached.

RCheesley commented:
I've run npm ci and regenerated assets (I'm in dev mode, locally) and still getting it opening in the old browser :( Also tried this in Gitpod to rule out funky stuff with my local environment.

@escopecz can you please try in Gitpod so we have reproducible environments with the sample data?

escopecz commented:
I don't think it's the old browser that's opening. I think it's the GrapesJS builder but having some issue with the content. I'll try to use the sample data as you do when I find a free minute.

matbcvo commented:
I can reproduce @RCheesley's issue on Gitpod with sample data loaded.

JavaScript error appears in the console when opening the Builder:
```javascript
Uncaught Error: No valid HTML template found
    at O.getOriginalContentHtml (builder.js?v05d545f1:10:2514)
    at tc.initEmailHtml (builder.js?v05d545f1:85:5826)
    at tc.initGrapesJS (builder.js?v05d545f1:85:3225)
    at Mautic.launchBuilder (builder.js?v05d545f1:85:9673)
    at HTMLButtonElement.onclick (1:1:8)
    at Object.trigger (jquery.js?v05d545f1:8635:27)
    at HTMLButtonElement.<anonymous> (jquery.js?v05d545f1:8707:17)
    at Function.each (jquery.js?v05d545f1:383:19)
    at jQuery.fn.init.each (jquery.js?v05d545f1:205:17)
    at jQuery.fn.init.trigger (jquery.js?v05d545f1:8706:15)
```


https://github.com/user-attachments/assets/0ef95d45-5538-42e2-919e-555f14b93f27

Also I tested it on the 6.x branch, encountered the same issue. The builder opens, but still broken and the same error appears in the console. I can't close the builder or save, it's stuck.

https://github.com/user-attachments/assets/4377a430-9cfa-4c54-94e1-cf49382b6500

matbcvo commented:
I tested it on the 6.x branch, encountered the same issue - also can't close builder or save, it's stuck.

https://github.com/user-attachments/assets/4377a430-9cfa-4c54-94e1-cf49382b6500

escopecz commented:
@RCheesley do you mind merging this and fixing the sample data issue later as it wasn't introduced in this PR?

RCheesley commented:
@escopecz  sure let's raise a separate issue so we don't lose track of it.

kuzmany reviewed (approved):
Finally 👍 
Seems good.
Works and code diff seems 👍

RCheesley commented:
Flagged up the issue in #14563 to be fixed separately.

RCheesley committed:
Merge branch '6.x' into removing-legacy-builder

RCheesley committed:
Merge branch '6.x' into removing-legacy-builder
